test/opt/test_pass_manager.cpp

```diff
@@ -91,4 +96,44 @@ TEST_F(PassManagerTest, Run) {
   RunAndCheck(text.c_str(), (text + "OpSource ESSL 310\nOpNop\n").c_str());
 }
 
+// A pass that appends an OpTypeVoid instruction that uses a given id.
+class AppendTypeVoidInstPass : public opt::Pass {
+ public:
+  AppendTypeVoidInstPass(uint32_t result_id) : result_id_(result_id) {}
+  const char* name() const override { return "AppendTypeVoidInstPass"; }
+  bool Process(ir::Module* module) override {
+    auto inst = MakeUnique<ir::Instruction>(SpvOpTypeVoid, 0, result_id_,
+                                            std::vector<ir::Operand>{});
+    module->AddType(std::move(inst));
+    return true;
+  }
+
+ private:
+  uint32_t result_id_;
+};
+
+TEST(PassManager, RecomputeIdBoundAutomatically) {
+  ir::Module module;
+  EXPECT_THAT(GetIdBound(module), Eq(0u));
+
+  opt::PassManager manager;
+  manager.Run(&module);
+  manager.AddPass<AppendOpNopPass>();
+  // With no ID changes, the ID bound does not change.
+  EXPECT_THAT(GetIdBound(module), Eq(0u));
+
+  // Now we force an Id of 100 to be used.
+  manager.AddPass(MakeUnique<AppendTypeVoidInstPass>(100));
+  EXPECT_THAT(GetIdBound(module), Eq(0u));
+  manager.Run(&module);
+  // The Id has been updated automatically, even though the pass
+  // did not update it.
+  EXPECT_THAT(GetIdBound(module), Eq(101u));
+
+  // Try one more time!
+  manager.AddPass(MakeUnique<AppendTypeVoidInstPass>(200));
+  manager.Run(&module);
+  EXPECT_THAT(GetIdBound(module), Eq(201u));
+}
```

Comment:
Probably we should try adding another `AppendTypeVoidInstPass` with a much smaller id to make sure that the id bound won't decrease?